p2p/src/network/pubsub/p2p_network_pubsub_reducer.rs

```diff
@@ -531,17 +669,63 @@ impl P2pNetworkPubsubState {
                     return;
                 };
                 if topic_state.on_mesh() {
-                    state.publish(&message)
+                    state.publish(message)
                 } else {
                     let ctr = state.message.control.get_or_insert_with(Default::default);
                     ctr.ihave.push(pb::ControlIHave {
                         topic_id: Some(message.topic.clone()),
-                        message_ids: message_id.clone().into_iter().collect(),
+                        message_ids: vec![message_id.to_raw_bytes()],
                     })
                 }
             });
+    }
 
-        Ok(())
+    /// Processes an incoming message by checking for duplicates and deserializing its contents.
+    ///
+    /// This function performs two main operations:
+    /// 1. Deduplication: Tracks recently seen messages using their signatures to avoid processing duplicates
+    /// 2. Deserialization: Converts valid message data into a `GossipNetMessageV2` structure
+    ///
+    /// # Arguments
+    ///
+    /// * `message` - The incoming message to process
+    /// * `seen_limit` - Maximum number of message signatures to keep in the deduplication cache
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(Some(GossipNetMessageV2))` - Successfully processed and deserialized message
+    /// * `Ok(None)` - Message was a duplicate (already seen)
+    /// * `Err(String)` - Error during processing (invalid message format or deserialization failure)
+    ///
+    #[inline(never)]
+    fn reduce_incoming_message(
+        &mut self,
+        message: &Message,
+        seen_limit: usize,
+    ) -> Result<Option<GossipNetMessageV2>, String> {
+        if let Some(signature) = &message.signature {
```

Comment:
I think this shouldn't ever be `None`, so it is safe to instead
```suggestion
       let Some(signature) = &message.signature else { bug_condition_and_return }
```

Reply:
I mean, the current version should work as is too, but this would make the intention clearer, now it looks like the message can come without signature, which is not the case